internal/pkg/cli/task_run.go

```diff
@@ -0,0 +1,270 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package cli
+
+import (
+	"errors"
+	"fmt"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/config"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/docker/dockerfile"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"regexp"
+	"strings"
+)
+
+var regions = []string{
+	"us-east-1",
+	"us-east-2",
+	"us-west-1",
+	"us-west-2",
+	"af-south-1",
+	"ap-east-1",
+	"ap-south-1",
+	"ap-northeast-1",
+	"ap-northeast-2",
+	"ap-northeast-3",
+	"ap-southeast-1",
+	"ap-southeast-2",
+	"ca-central-1",
+	"cn-north-1",
+	"cn-northwest-1",
+	"eu-central-1",
+	"eu-west-1",
+	"eu-west-2",
+	"eu-south-1",
+	"eu-west-3",
+	"eu-north-1",
+	"me-south-1",
+	"sa-east-1",
+}
+
+type runTaskVars struct {
+	*GlobalOpts
+	Num    uint8
+	CPU    uint16
+	Memory uint16
+
+	Image          string
+	DockerfilePath string
+
+	TaskRole string
+
+	SubnetID         string
+	SecurityGroupIDs []string
+	App              string
+	Env              string
+
+	EnvVars  map[string]string
+	Commands string
+}
+
+type runTaskOpts struct {
+	runTaskVars
+
+	// Interfaces to interact with dependencies.
+	fs    afero.Fs
+	store store
+	df    dockerfileParser
+
+	// sets up Dockerfile parser using fs and input path
+	setupParser func(opts *runTaskOpts)
+}
+
+func newTaskRunOpts(vars runTaskVars) (*runTaskOpts, error) {
+	store, err := config.NewStore()
+	if err != nil {
+		return nil, fmt.Errorf("new config store: %w", err)
+	}
+
+	return &runTaskOpts{
+		runTaskVars: vars,
+
+		fs:    &afero.Afero{Fs: afero.NewOsFs()},
+		store: store,
+
+		setupParser: func(o *runTaskOpts) {
+			o.df = dockerfile.New(o.fs, o.DockerfilePath)
+		},
+	}, nil
+}
+
+// Validate returns an error if the flag values passed by the user are invalid.
+func (o *runTaskOpts) Validate() error {
+	if o.Num == 0 {
+		return errors.New("number of tasks must be positive")
+	}
+
+	if o.CPU == 0 {
+		return errors.New("CPU units must be positive")
+	}
+
+	if o.Memory == 0 {
+		return errors.New("memory must be positive")
+	}
+
+	if o.Image != "" && o.DockerfilePath != "" {
+		return errors.New("cannot specify both image and dockerfile path")
+	}
+
+	if o.Image != "" {
+		if err := o.validateImageName(); err != nil {
+			return err
+		}
+	}
+
+	if o.DockerfilePath != "" {
+		if _, err := o.fs.Stat(o.DockerfilePath); err != nil {
+			return err
+		}
+	}
```

Comment:
nit: This could potentially be an if .. else-if block since you can't have both Image and DockerfilePath specified.

Reply:
@pkandasamy91 that should be taken care of already in [line 81](https://github.com/aws/amazon-ecs-cli-v2/pull/1015/files#diff-d83adc990a0da087a39c02967f267ed3R81)